src/helpers.js

```diff
@@ -1,19 +1,20 @@
 const { APP_NAME, EXPIRY_PERIOD } = require('./constants');
 const { cli } = require('cli-ux');
 const { CLIError } = require('@oclif/errors');
+const { customAlphabet } = require('nanoid');
 const fs = require('fs');
 const { getListOfFunctionsAndAssets } = require('@twilio-labs/serverless-api/dist/utils/fs');
 const moment = require('moment');
 const path = require('path');
 const { TwilioServerlessApiClient } = require('@twilio-labs/serverless-api');
 
-function getPin() {
-  return Math.floor(Math.random() * 900000) + 100000;
+function getRandomInt(length) {
+  return customAlphabet('1234567890', length)();
```

Comment:
The npm nanoid module you're using has a [probability calculator](https://zelark.github.io/nano-id-cc/). Have we tried putting our parameters (custom alphabet, length, rate of serverless app creation) and see what is the collision probability? I'm curious at what point do we have to update this implementation again.

Reply:
This is a fun question!

So I think there are two figures to look at here:
- Probability of one collision occurring within all deployments (the [birthday problem](https://en.wikipedia.org/wiki/Birthday_problem))
- Probability that a single deployment will result in collision (# of deployments / available domain names)

With the current version of the plugin, each deployment is identified with 4 random digits. This means that there will be a 50% chance of one collision occurring after only 118 deployments.

We are seeing about 100 new deployments each week, and we currently have ~3000 deployments. 

So with 3000 deployments and a 4 digit app identifier, we have a 100% chance that a collision has occurred and a 30% chance that a single deployment will result in a collision. Not good. 

With the new version of the plugin, each deployment is identified with 8 random digits. This means that there will be a 50% chance of one collision occurring after 11,775 deployments. 

With 25,000 deployments (about 100 per week for five years), there will be a 95.6% chance that one collision will have occurred, and a 1 in 4000 chance that a single deployment will result in a collision. 

Even after 100,000 deployments, there will be a 1 in 1000 chance that a single deployment will collide with an existing one. Much better than the current odds of 1 in 3. 

I made a calculator here if you want to play with the numbers: https://53cir.csb.app/

Reply:
Nice very detailed! Definitely good for now with our current traffic and short term goal for signal.